pkg/smokescreen/smokescreen_test.go

```diff
@@ -443,11 +443,18 @@ func TestInvalidHost(t *testing.T) {
 }
 
 var hostSquareBracketsCases = []struct {
-	scheme    string
-	proxyType string
+	scheme         string
+	proxyType      string
+	hostname       string
+	decisionReason string
 }{
-	{"http", "http"},
-	{"https", "connect"},
+	{"http", "http", "[stripe.com]", "host matched rule in global deny list"},
+	{"https", "connect", "[stripe.com]", "host matched rule in global deny list"},
+	{"http", "http", "[[stripe.com]]", "host matched rule in global deny list"},
+	{"https", "connect", "[[stripe.com]]", "host matched rule in global deny list"},
+	{"http", "http", "[[[stripe.com]]]", "host matched rule in global deny list"},
+	{"https", "connect", "[[[stripe.com]]]", "Destination host cannot be determined"},
```

Comment:
I don't know why this happens, but I assume it's got something to do with how Go constructs the CONNECT request. These should both be denied anyway, so I wasn't too concerned about the specific error